pkg/storage/replica.go

```diff
@@ -4147,50 +4195,50 @@ func (r *Replica) executeBatch(
 		if pErr != nil {
 			switch tErr := pErr.GetDetail().(type) {
 			case *roachpb.WriteTooOldError:
-				// WriteTooOldErrors may be the product of raft replays. If
-				// timestamp of the request matches exactly with the existing
-				// value, maybe propagate the WriteTooOldError to let client
-				// retry at a higher timestamp. Keep in mind that this replay
-				// protection is best effort only. If replays come out of
-				// order, we'd expect them to succeed as the timestamps which
-				// would match on a successive replay won't match if the replay
-				// is delivered only after another raft command has been applied
-				// to the same key.
-				if ba.Timestamp.Next().Equal(tErr.ActualTimestamp) {
-					// If in a txn, propagate WriteTooOldError immediately. In
-					// a txn, intents from earlier commands in the same batch
-					// won't return a WriteTooOldError.
-					if ba.Txn != nil {
-						return nil, result, pErr
-					}
-					// If not in a txn, need to make sure we don't propagate the
-					// error unless there are no earlier commands in the batch
-					// which might have written the same key.
-					var overlap bool
+				// We got a WriteTooOldError. In case this is a transactional request,
```

Comment:
:.(

I thought we were getting rid of this code block. Why do you think this is necessary? Your explanation looks to me like a premature optimization. Do we have some case in mind where this is crucial for performance? My intuition is the complexity here is a mistake given the avowed justification.